### DISCUSSION TITLE
JRuby team is interested in learning more about Commonhaus

### DISCUSSION THREAD

**Discussion by headius** [commonhaus/foundation#146] [Joining Commonhaus]
### Project information

- **Project name**: JRuby, an implementation of Ruby on the JVM
- **Project website**: [JRuby.org](https://www.jruby.org/)
- **Code repository**: https://github.com/jruby/jruby
- **License**: Eclipse Public License 2..0 (optionally LGPL 2 or GPL 2)

### Describe your project

JRuby is the most used, most successful alternative implementation of Ruby in the world. Over the past 20 years, we have put out dozens of releases and helped drive forward the future of JVM languages. JRuby touches on many areas of tech: language parsing, character and text encodings, regular expressions, native function and memory access, concurrency including virtual threading, JIT compilation using JVM bytecode, optimization using invokedynamic and JVM profiling tools, and maintaining strong compatibility with current versions of Ruby.

Users of JRuby deploy Rails and other applications at large scale, handling millions of requests per day all over the world. JRuby has saved many Ruby shops that needed better scaling or features than the standard implementation.

We have always had a fully open project and process. JRuby developers communicate on the open source Matrix chat network, and all our bugs and discussions are fully open on Github. In addition, we have worked hard to visit Ruby and JVM conferences around the world to share the work we are doing and collaborate with users. JRuby is a critical part of the Ruby ecosystem and a prominent member of the JVM language community.

JRuby would be a perfect match for Commonhaus for several reasons:

* Keeping up with Ruby compatibility takes a lot of effort and expertise on the language and how it is implemented.
* New technologies in the JVM push JRuby forward every year, and integrating them is a big and rewarding job.
* As the only production-deployed alternative Ruby, JRuby provides a whole new set of deployment and scaling opportunities for Ruby developers.
* Our users have been very satisfied with the project and frequently ask how they can contribute resources or funds to keep the project going. Without an organization for JRuby to live within, we have had to decline their generousity.
* Current sponsorship of JRuby is likely to come to and end this year, which will severely impact our ability to keep up with Ruby releases and customer needs.

**Comment by ebullient** [comment]
We're still working on bootstrapping logistics, but here are the steps. ;)

- [ ] **Review the bylaws**, especially sections about [projects](https://www.commonhaus.org/bylaws/cf-membership.html#projects), [decision making](https://www.commonhaus.org/bylaws/decision-making.html), and [voting for significant decisions](https://www.commonhaus.org/bylaws/cf-council.html#voting). You may have seen some PRs go through as I work with the current EGC to create a record of approval with the newly established EGC) to ratify them and you're welcome to weigh in on these.
- [ ] **Read the [bootstrapping agreement](https://github.com/commonhaus/foundation/blob/main/agreements/bootstrapping/bootstrapping.md)**.
- [ ] If all is well, and you agree to participate in the bootstrapping phase to help us build this thing, then **create a PR** with the following:

    1. A filled out copy of the bootstrapping agreement. You can make a copy of [bootstrapping.md](https://github.com/commonhaus/foundation/blob/main/agreements/bootstrapping/bootstrapping.md) file and check that in, or checkin a filled out [pdf](https://github.com/commonhaus/foundation/releases/download/final-draft/bootstrapping.pdf). Please follow naming conventions.
    2. An updated [CONTACTS.yaml](https://github.com/commonhaus/foundation/blob/main/CONTACTS.yaml) that names your project representative in the EGC section
    3. An updated [PROJECTS.yaml](https://github.com/commonhaus/foundation/blob/main/PROJECTS.yaml) that provides details / describes your project
    4. An updated [TRADEMARKS.md](https://github.com/commonhaus/foundation/blob/main/TRADEMARKS.md) that adds your project name and project logo (if you have one). Please include the footnote reference, as we are using your project name and logo with your permission until we have a formal agreement in place and resources are transferred.

    The commit should be "Signed-off-by" your GH login (`git commit -s -m "<project> bootstrapping agreement, yyyy-mm-dd"`)

**Comment by kenfinnigan** [comment]
@headius if you want to meet to discuss the foundation, or if there's further information you need to evaluate whether Commonhaus is the right fit for JRuby, please reach out
